server/test/app/ApiAuthenticationTest.java

```diff
@@ -0,0 +1,84 @@
+package app;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static play.api.test.Helpers.testServerPort;
+import static play.test.Helpers.fakeRequest;
+import static play.test.Helpers.route;
+
+import auth.ApiKeyGrants;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Optional;
+import models.ApiKey;
+import org.junit.Before;
+import org.junit.Test;
+import org.pac4j.core.context.HttpConstants;
+import play.mvc.Http;
+import play.mvc.Result;
+import repository.ResetPostgres;
+
+/**
+ * Test for asserting path-based authentication logic as specified by pac4j.security.rules value in
+ * application.conf
+ */
+public class ApiAuthenticationTest extends ResetPostgres {
+  private static final String keyId = "keyId";
+  private static final String secret = "secret";
+
+  @Before
+  public void setUp() {
+    ApiKey apiKey = resourceCreator.createActiveApiKey("test-key", keyId, secret);
+    apiKey.getGrants().grantProgramPermission("mock-program", ApiKeyGrants.Permission.READ);
+    apiKey.save();
+  }
+
+  @Test
+  public void nonApiRoutes_doNotRequireApiAuth() {
+    Result result =
+        doGetRequest(
+            fakeRequest(
+                "GET", controllers.routes.HomeController.loginForm(Optional.empty()).url()));
+    System.out.println(result.headers());
+    assertThat(result.status()).isEqualTo(HttpConstants.OK);
+  }
+
+  @Test
+  public void apiRoutes_success() {
+    String rawCredentials = keyId + ":" + secret;
+    String creds =
+        Base64.getEncoder().encodeToString(rawCredentials.getBytes(StandardCharsets.UTF_8));
+
+    Result result =
+        doGetRequest(
+            fakeRequest(
+                    "GET",
+                    controllers.api.routes.ProgramApplicationsApiController.list("mock-program")
+                        .url())
+                .remoteAddress("1.1.1.1")
+                .header("Authorization", "Basic " + creds));
+
+    assertThat(result.status()).isEqualTo(HttpConstants.OK);
+  }
+
+  @Test
+  public void apiRoutes_invalidKey() {
```

Comment:
Do we have an equivalent test for UNAUTHORIZED status for non-API paths?

Reply:
yep SecurityBrowserTest